plugins/slipping-away-users/frontend/public/javascripts/countly.views.js

```diff
@@ -1,296 +1,154 @@
-            var item = slippingData[i],
-                index = i;
-
-            slippingChartData.chartDP.dp[0].data.push([index, item.count]);
-            slippingChartData.chartDP.ticks.push([index, item.period + ' days']);
-            item.percentage = "<div style='width:90%;  box-sizing:border-box;'><div class='percent-bar' style='width:" + (item.percentage) + "%;'></div><span style='margin-right:-70px;'>" + item.percentage + "%</span></div>";
-            if (countlyGlobal.plugins.indexOf("users") >= 0) {
-                if (item.count > 0) {
-                    item.userList = "<a class='extable-link table-link green'  href='#/users/query/" + JSON.stringify({"lac": {"$lt": item.timeStamp}}) + "'   target='_blank'>" +
-                "<i class='material-icons'>open_in_new</i></a>" +
-                '<a class="table-link green external"   data-localize="userdata.list" onclick="slippingDataOnClick(' + item.timeStamp + ')">View User List</a> ' ;
-                }
-                else {
-                    item.userList = '<span style="float: right;float: right;margin-right: 23px;">' + jQuery.i18n.map['slipping.no-user'] + '</span>';
-                }
-            }
-        }
-
-        slippingChartData.chartDP.dp[0].data.push([slippingData.length, null]);
-
-        this.templateData = {
-            "page-title": jQuery.i18n.map["slipping.title"],
-            "font-logo-class": "fa-sign-out",
-            "chart-helper": "durations.chart",
-            "table-helper": "durations.table",
-            "drill-filter": countlyGlobal.plugins.indexOf('drill') >= 0,
-        };
-        if (!isRefresh) {
-            $(this.el).html(this.template(this.templateData));
-            countlyCommon.drawGraph(slippingChartData.chartDP, "#dashboard-graph", "bar");
-            $(" <div class='graph-description'>" + jQuery.i18n.map["slipping.graph-description"] + "</div>").insertAfter("#view-filter");
+        },
+    },
+    methods: {
+        onApplyFilter: function() {
+            this.$store.dispatch("countlySlippingAwayUsers/fetchSlippingAwayUsers");
+        },
+    }
+});
 
-            var columnsDefine = [
-                { "mData": "period", sType: "numeric", "sTitle": jQuery.i18n.map["slipping.period"], "sWidth": "20%" },
-                {
-                    "mData": "count",
-                    "sType": "numeric",
-                    "sWidth": "20%",
-                    "sTitle": jQuery.i18n.map["slipping.count"],
-                    "mRender": function(d) {
-                        return countlyCommon.formatNumber(d);
+var SlippingAwayUsersBarChart = countlyVue.components.BaseComponent.extend({
+    template: "#slipping-away-users-bar-chart",
+    data: function() {
+        return {};
+    },
+    computed: {
+        slippingAwayUsers: function() {
+            return this.$store.state.countlySlippingAwayUsers.slippingAwayUsers;
+        },
+        slippingAwayUsersOptions: function() {
```

Comment:
Current cly-chart implementation offers almost no abstractions in terms of axes and other configuration. Once chart implementation is finalized we might need to get back to this. @itsiprikshit